api/client/login.go

```diff
@@ -74,6 +74,10 @@ func (cli *DockerCli) CmdLogin(args ...string) error {
 			username = authconfig.Username
 		}
 	}
+
+	if username == "" {
+		return fmt.Errorf("Error: Non-null Username Required")
+	}
```

Comment:
Hi @duglin. We got value of username(in line 69) from command parameters, and have trim space( line 72) in this case.  So do we need trim  again, or I miss something?